rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java

```diff
@@ -1090,30 +1067,30 @@ private Federation deserializeFederationAccordingToVersion(
         int version,
         BridgeConstants bridgeConstants
     ) {
-        switch (version) {
-            case LEGACY_ERP_FEDERATION_FORMAT_VERSION:
-                return BridgeSerializationUtils.deserializeLegacyErpFederation(
-                    data,
-                    bridgeConstants,
-                    activations
-                );
-            case P2SH_ERP_FEDERATION_FORMAT_VERSION:
-                return BridgeSerializationUtils.deserializeP2shErpFederation(
-                    data,
-                    bridgeConstants
-                );
-            case STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION:
-                return BridgeSerializationUtils.deserializeStandardMultisigFederation(
-                    data,
-                    networkParameters
-                );
-            default:
-                // To keep backwards compatibility
-                return BridgeSerializationUtils.deserializeStandardMultisigFederation(
-                    data,
-                    networkParameters
-                );
+        if (version == STANDARD_MULTISIG_FEDERATION.getFormatVersion()) {
+            return BridgeSerializationUtils.deserializeStandardMultisigFederation(
+                data,
+                networkParameters
+            );
         }
+        if (version == NON_STANDARD_ERP_FEDERATION.getFormatVersion()) {
+            return BridgeSerializationUtils.deserializeLegacyErpFederation(
+                data,
+                bridgeConstants,
+                activations
+            );
+        }
+        if (version == P2SH_ERP_FEDERATION.getFormatVersion()) {
+            return BridgeSerializationUtils.deserializeP2shErpFederation(
+                data,
+                bridgeConstants
+            );
+        }
+        // To keep backwards compatibility
+        return BridgeSerializationUtils.deserializeStandardMultisigFederation(
+            data,
+            networkParameters
+        );
     }
```

Comment:
I believe we should move this method out of the bridge storage provider to reduce the amount of code in this class. The same applies to `BridgeSerializationUtils`. We could have two one class for serialization and a second one for deserialization. Let me know what you think.

Reply:
Agree but I think is better to create another task for that